src/Corax/IndexWriter.cs

```diff
@@ -364,6 +371,30 @@ void ExactInsert(ReadOnlySpan<byte> value)
 
                 if (binding.HasSuggestions)
                     AddSuggestions(binding, slice);
+
+                scope?.Dispose();
+            }
+        }
+
+        private static unsafe ByteStringContext<ByteStringMemoryCache>.InternalScope CreateNormalizedTerm(ByteStringContext context, ReadOnlySpan<byte> value, out Slice slice)
```

Comment:
This is an intermediate for writing always. We need a few tests that trigger this and ensure that we can do exact searches over those terms. @maciejaszyk can you follow this tomorrow?

Also eventually we will need to do some work to unify the buffers. Lately we have been trafficking buffers all over the place with @ayende and we are in a very weak error prone situation from the reliability front. For experimental it is OK, but we need a more robust solution for it. https://issues.hibernatingrhinos.com/issue/RavenDB-18860